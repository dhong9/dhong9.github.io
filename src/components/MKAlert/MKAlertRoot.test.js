import renderer from 'react-test-renderer';
import { ThemeProvider, createTheme } from '@mui/material/styles';
import MKAlertRoot from 'components/MKAlert/MKAlertRoot';
import pxToRem from "assets/theme/functions/pxToRem";
import linearGradient from "assets/theme/functions/linearGradient";

describe('MKAlertRoot', () => {
    it('renders', () => {
        // Create a dummy theme object with the mock pxToRem
        const theme = createTheme({
            palette: {
                white: {
                    main: "#00F"
                },
                gradients: {
                    "#FF00FF": "#FF00FF"
                }
            },
            typography: {
                size: {
                    xl: "16pt"
                },
                fontWeightMedium: 500
            },
            borders: {
                borderRadius: 10
            },
            functions: { pxToRem, linearGradient }
        });
        const ownerState = {
            color: "#FF00FF"
        };

        const component = renderer.create(
            <ThemeProvider theme={theme}>
                <MKAlertRoot onClick={jest.fn()} ownerState={ownerState}>Hello!</MKAlertRoot>
            </ThemeProvider>
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
