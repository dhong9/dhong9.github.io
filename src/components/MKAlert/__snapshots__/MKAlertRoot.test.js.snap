// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MKAlertRoot renders with custom color 1`] = `
<div
  className="MuiBox-root css-opofj1"
  onClick={[MockFunction]}
>
  Hello!
</div>
`;

exports[`MKAlertRoot renders without custom color 1`] = `
<div
  className="MuiBox-root css-16qid04"
  onClick={[MockFunction]}
>
  Hello!
</div>
`;
